fili-core/src/main/java/com/yahoo/bard/webservice/data/config/table/BaseTableLoader.java

```diff
@@ -77,24 +76,21 @@ protected BaseTableLoader(DateTimeZone defaultTimeZone, DataSourceMetadataServic
      * Builds and loads the physical tables for the physical table definitions as well.
      *
      * @param apiMetrics  The set of metric names surfaced to the api
-     * @param druidMetrics  Names of druid datasource metric columns
      * @param tableDefinitions  A list of config objects for physical tables
      * @param dictionaries  The container for all the data dictionaries
      *
      * @return A table group binding all the tables for a logical table view together.
      */
     public TableGroup buildDimensionSpanningTableGroup(
             Set<ApiMetricName> apiMetrics,
-            Set<FieldName> druidMetrics,
             Set<PhysicalTableDefinition> tableDefinitions,
             ResourceDictionaries dictionaries
     ) {
         // Load a physical table for each of the table definitions
-        LinkedHashSet<PhysicalTable> physicalTables = new LinkedHashSet<>();
-        for (PhysicalTableDefinition def : tableDefinitions) {
-            PhysicalTable table = loadPhysicalTable(def, druidMetrics, dictionaries);
-            physicalTables.add(table);
-        }
+        LinkedHashSet<PhysicalTable> physicalTables = loadPhysicalTablesWithDependency(
+                tableDefinitions,
+                dictionaries
+        );
```

Comment:
Per discussion yesterday, for this to work we need to have both the list of table names or table definitions which define the table group and ALSO have the total set of table definitions for building dependent tables.